dev/scaletesting/syntheticgithubdata/oauth.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"sync/atomic"
+
+	"github.com/google/go-github/v41/github"
+
+	"github.com/sourcegraph/sourcegraph/lib/group"
+)
+
+// generateUserOAuthCsv creates user impersonation OAuth tokens and writes them to a CSV file together with the usernames.
+func generateUserOAuthCsv(ctx context.Context, users []*user, tokensDone int64) {
+	tg := group.NewWithResults[userToken]().WithMaxConcurrency(1000)
+	for _, u := range users {
+		currentU := u
+		tg.Go(func() userToken {
+			token := currentU.executeCreateImpersonationToken(ctx)
+			atomic.AddInt64(&tokensDone, 1)
+			progress.SetValue(5, float64(tokensDone))
+			return userToken{
+				login: currentU.Login,
+				token: token,
+			}
+		})
+	}
+	pairs := tg.Wait()
+
+	csvFile, err := os.Create("users.csv")
+	defer csvFile.Close()
```

Comment:
```suggestion
	defer func() {
		cerr := csvFile.Close()
		if err == nil {
			err = cerr
		}
	}()
```
This should resolve the complaining linting error. I believe the difference here on why it complains is that this is a writable file that is executing writing commands that take longer to complete than just reading and loading from those files, so it may not require a check for errors at Close() time